cmdget.go

```diff
@@ -330,3 +345,81 @@ func removeAutoGoMod(m *moduleVCSInfo) (bool, error) {
 func autoGoMod(mpath string) string {
 	return "// Generated by gohack; DO NOT EDIT.\nmodule " + mpath + "\n"
 }
+
+// checkCanReplace checks whether it may be possible to replace
+// the module in the given go.mod file.
+func checkCanReplace(f *modfile.File, mod string) error {
+	var found *modfile.Replace
+	for _, r := range f.Replace {
+		if r.Old.Path != mod {
+			continue
+		}
+		if found != nil {
+			return errors.Newf("found multiple existing replacements for %q", mod)
+		}
+		if r.New.Version == "" {
+			return errors.Newf("%s is already replaced by %s", mod, r.New.Path)
+		}
+	}
+	return nil
+}
+
+// replaceModule adds or modifies a replace statement in f for mod
+// to be replaced with dir.
+func replaceModule(f *modfile.File, mod string, dir string) error {
+	var found *modfile.Replace
+	for _, r := range f.Replace {
+		if r.Old.Path != mod {
+			continue
+		}
+		// These checks shouldn't fail when checkCanReplace has been
+		// called previously, but check anyway just to be sure.
+		if found != nil {
+			return errors.Newf("found multiple existing replacements for %q", mod)
+		}
+		if r.New.Version == "" {
+			return errors.Newf("%s is already replaced by %s", mod, r.New.Path)
+		}
+		found = r
+	}
+	if found == nil {
+		// No existing replace statement. Just add a new one.
+		if err := f.AddReplace(mod, "", dir, ""); err != nil {
+			return errors.Wrap(err)
+		}
+		return nil
```

Comment:
If `errors.Wrap` knew to keep nil errors as nil, you'd be able to just do a single-line return here I think.

Though it would be a bit of an ugly line.

Reply:
drive-by comment, sorry. The more popular `pkg/errors` handles this case :)
https://github.com/pkg/errors/blob/c059e472/errors.go#L180-L183

Reply:
errors.Wrap does know to keep nil errors as nil, but it made for a bit of an ugly line so I avoided it here :)